File: base/protocol/DoubleItem.php
<?php

namespace FFan\Dop\Protocol;

use FFan\Dop\Exception;

/**
 * Class DoubleItem
 * @package FFan\Dop\Protocol
 */
class DoubleItem extends Item
{
    /**
     * @var int 类型
     */
    protected $type = ItemType::DOUBLE;

    /**
     * 设置默认值
     * @param string $value
     * @throws Exception
     */
    public function setDefault($value)
    {
        if (!preg_match('/^(-?\d+)(\.\d+)?$/', $value)) {
            throw new Exception($value . ' 不能作为 double 默认值');
        }
        $this->default = $value;
    }
}

File: test/java/DopTest/GsonTest/src/com/ffan/dop/DoubleItem.java
package com.ffan.dop;

class DoubleItem extends Item {
    /**
     * 值
     */
    private double value;

    /**
     * 构造函数
     */
    DoubleItem(double value) {
        this.value = value;
        this.type = ItemType.DOUBLE_TYPE;
    }

    /**
     * 获取值 String
     */
    public String getValueString() {
        return String.valueOf(this.value);
    }

    /**
     * 获取值 int
     */
    public long getValueInt() {
        return Math.round(this.value);
    }

    /**
     * 获取值double
     */
    public double getValueDouble() {
        return this.value;
    }

    /**
     * 获取值 bool
     */
    public boolean getValueBool() {
        return 0 != this.getValueInt();
    }
}